tests/test_benchmark_free_energy.py

```diff
@@ -0,0 +1,98 @@
+    elapsed_ns = time.perf_counter_ns() - start_time
+    return result, elapsed_ns
+
+
+hif2a_single_topology_leg_params = {
+    f"{host_name}-{n_windows}": (host_name, n_windows)
+    for host_name, n_windows in product([None, "complex", "solvent"], [5, 10, 20])
+}
+
+
+@pytest.fixture(
+    scope="module", params=hif2a_single_topology_leg_params.values(), ids=tuple(hif2a_single_topology_leg_params.keys())
+)
+def hif2a_single_topology_leg(request):
+    host_name, n_windows = request.param
+    forcefield = Forcefield.load_default()
+    host_config: Optional[HostConfig] = None
+
+    if host_name == "complex":
+        with resources.path("timemachine.testsystems.data", "hif2a_nowater_min.pdb") as protein_path:
+            host_sys, host_conf, box, _, num_water_atoms = builders.build_protein_system(
+                str(protein_path), forcefield.protein_ff, forcefield.water_ff
+            )
+            box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+        host_config = HostConfig(host_sys, host_conf, box, num_water_atoms)
+    elif host_name == "solvent":
+        host_sys, host_conf, box, _ = builders.build_water_system(4.0, forcefield.water_ff)
+        box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+        host_config = HostConfig(host_sys, host_conf, box, host_conf.shape[0])
+
+    mol_a, mol_b, core = get_hif2a_ligand_pair_single_topology()
+    single_topology = SingleTopology(mol_a, mol_b, core, forcefield)
+    host = setup_optimized_host(single_topology, host_config) if host_config else None
+
+    lambdas = np.linspace(0.0, 0.2, n_windows)
+
+    initial_states = setup_initial_states(single_topology, host, DEFAULT_TEMP, lambdas, seed=2023, min_cutoff=0.7)
+
+    return host_name, n_windows, initial_states
+
+
+@pytest.mark.nightly(reason="Slow")
+@pytest.mark.parametrize("enable_hrex", [False, True])
+def test_benchmark_hif2a_single_topology(hif2a_single_topology_leg, enable_hrex):
+
+    host_name, n_windows, initial_states = hif2a_single_topology_leg
+
+    n_frames = 1000 // n_windows
+    md_params = MDParams(n_frames=n_frames, n_eq_steps=1, steps_per_frame=400, seed=2023)
+    temperature = DEFAULT_TEMP
+
+    if enable_hrex:
+        run = partial(
+            run_sims_hrex,
+            initial_states,
+            md_params,
+            n_frames_per_iter=5,
+            temperature=temperature,
+            print_diagnostics_interval=None,
+        )
+    else:
+        run = partial(run_sims_sequential, initial_states, md_params, temperature=temperature, keep_idxs=[])
+
+    _, elapsed_ns = run_with_timing(run)
+
+    print("hrex:", enable_hrex)
+    print("host:", host_name)
+    print("n_windows:", n_windows)
+    print("n_frames:", n_frames)
+
+    dt_ns = initial_states[0].integrator.dt / 1e3
+    ns = md_params.n_frames * md_params.steps_per_frame * dt_ns
+    elapsed_days = elapsed_ns / 1e9 / 60 / 60 / 24
+
+    print(f"total ns/day: {n_windows * ns / elapsed_days:.1f}")
```

Comment:
nit: locally, this is printing ns/day numbers far in excess of what I'd expect, e.g.

```
...
hrex: False
host: None
n_windows: 20
n_frames: 50
total ns/day: 6320.9
...
hrex: True
host: None
n_windows: 20
n_frames: 50
total ns/day: 3668.2
...
```

where I'd expect at most ~800 ns/day (based on timings from `test_benchmark.py`, `hif2a-rbfe-barostat-interval-15`)

but I don't immediately see an error. Is this expected to exceed the speed of running MD in a single window (i.e. without overheads from multiple contexts + replica-mixing)?

Reply:
Is this in excess of what we'd expect for a vacuum simulation? (Note `host: None` here)

Reply:
doh! sorry, overlooked that. not in excess of expectations for vacuum simulation, was mistakenly comparing to hif2a-rbfe benchmark script (with n_atoms=8840)